platform/linux-generic/include/odp/api/plat/ticketlock_inlines.h

```diff
@@ -47,8 +51,15 @@ _ODP_INLINE void odp_ticketlock_lock(odp_ticketlock_t *ticketlock)
 
 	/* Spin waiting for our turn. Use load-acquire so that we acquire
 	 * all stores from the previous lock owner */
-	while (ticket != odp_atomic_load_acq_u32(&ticketlock->cur_ticket))
-		odp_cpu_pause();
+	#ifdef CONFIG_WFE
+		sevl();
+		while (wfe() && ticket !=
+			monitor32((uint32_t *)&ticketlock->cur_ticket, __ATOMIC_ACQUIRE)) {
+		}
+	#else
+		while (ticket != odp_atomic_load_acq_u32(&ticketlock->cur_ticket))
+			odp_cpu_pause();
+	#endif
```

Comment:
Aarch64 specific code belongs under: `platform/linux-generic/arch/aarch64`. One potential solution would be to add a similar wrapper function (e.g. `_odp_ticketlock_lock()`) that we use for atomic functions.
See for example:
* `platform/linux-generic/include/odp/api/plat/atomic_inlines.h`
*  `platform/linux-generic/arch/default/odp/api/abi/atomic_generic.h`
* `platform/linux-generic/arch/aarch64/odp/api/abi/atomic_inlines.h`.

Reply:
Updated in V4.